test/functional/mempool_persist.py

```diff
@@ -85,5 +91,29 @@ def run_test(self):
         self.nodes.append(self.start_node(0, self.options.tmpdir))
         wait_until(lambda: len(self.nodes[0].getrawmempool()) == 5)
 
+        regtestdir0 = os.path.join(self.options.tmpdir, 'node0', 'regtest')
+        mempooldat0 = os.path.join(regtestdir0, 'mempool.dat')
+        regtestdir1 = os.path.join(self.options.tmpdir, 'node1', 'regtest')
+        mempooldat1 = os.path.join(regtestdir1, 'mempool.dat')
+        self.log.debug("Remove the mempool.dat file. Verify that dumpmempool to disk via RPC re-creates it")
+        os.remove(mempooldat0)
+        assert self.nodes[0].dumpmempool()
+        assert os.path.isfile(mempooldat0)
+
+        self.log.debug("Stop nodes, make node1 use mempool.dat from node0. Verify it has 5 transactions")
+        self.stop_nodes()
+        os.rename(mempooldat0, mempooldat1)
+        self.nodes.append(self.start_node(1, self.options.tmpdir))
+        # Give bitcoind a second to reload the mempool
+        time.sleep(1)
```

Comment:
No need for `time.sleep()` if using `wait_until()` below.

Reply:
I thought so too (and indeed it works without time.sleep()) - however the test has that as a pattern and I think we either remove it for the rest of the calls (at least within the same test) or it's inconsistent.

Reply:
I'm all for trying to match local style, but there's no benefit in copying bad patterns.

The `time.sleep` in L76 should be removed (and the `assert_equal()` in L78 replaced with a `wait_until()`) in a different PR.

The `time.sleep` in L85 serves a purpose - the mempool takes a while to load so we want to give it time before asserting it's empty. There's probably a better way of testing this, but we can't just replace the `assert_equal()` with a `wait_until()`.